make it possible to grep/awk/sed whole dirs including compressed files

InformatiQ opened the pull request.

InformatiQ committed:
make it possible to grep/awk/sed whole dirs invlusing compressed files

iksaif commented on CONTRIBUTE.txt:
```diff
@@ -0,0 +1,9 @@
+# create a virtualenv
+source bin/activarte
```
activate

iksaif commented on tailon/commands.py:
```diff
@@ -85,3 +92,19 @@ def tail_sed(self, n, fn, script, stdout, stderr):
         sed = self.sed(script, None, stdout=STREAM, stderr=STREAM, stdin=tail.stdout)
         tail.stdout.close()
         return tail, sed
+# app wide
```
?

InformatiQ committed:
add /dirs route to provide UI with dirs list to fill the selector

Commit:
Changed default file selector

Added option for using URL GET argument to select a default file to observ

Commit:
Changed default file selector

Added option for using URL GET argument to select a default file to observ

Commit:
Generated file from TS source

Commit:
Generated file from TS source

InformatiQ committed:
add dir-mode option to enable/disable running commands dir wide

Commit:
added new setting option handler to look for dir

Commit:
added new setting option handler to look for dir

InformatiQ committed:
add dir-mode option to enable/disable running commands dir wide

Commit:
Updated websocket request with dir_mode

Commit:
Updated websocket request with dir_mode

Commit:
Changed var name

Commit:
Changed var name

Commit:
Generated Js for previous changes

Commit:
Generated Js for previous changes

iksaif commented on CONTRIBUTE.txt:
```diff
@@ -0,0 +1,9 @@
+# or use webscoket client to query the backend
+# ws://localhost:8080/ws/917/ez2zsmvz/websocket
+# ["{\"command\":\"awk\",\"path\":\"/var/log/\",\"script\":\"services\",\"tail-lines\":10, \"dir_mode\": \"true\"}"]
+# I use soimple websocket browser plugin
```
simple

iksaif commented on CONTRIBUTE.txt:
```diff
@@ -0,0 +1,9 @@
+pip install .
+tailon -f /var/log -b localhost:8080 -d
+# either use the webUI on localhost:8080
+# or use webscoket client to query the backend
```
websocket

iksaif commented on tailon/commands.py:
```diff
@@ -85,3 +92,18 @@ def tail_sed(self, n, fn, script, stdout, stderr):
         sed = self.sed(script, None, stdout=STREAM, stderr=STREAM, stdin=tail.stdout)
         tail.stdout.close()
         return tail, sed
+    def all_awk(self, fn, script, stdout, stderr):
```
add line return before

geobeau committed:
Merge pull request #1 from criteo-forks/dir_mode_and_compressed

Dir mode and compressed

geobeau committed:
changed default action to grep in dir-mode

geobeau committed:
Changed display of dirname to remove path

geobeau committed:
Revert server.py to 87b4ce4c359848287d28e1b421d12399bc2e716b

geobeau committed:
Changed to resolve conflict

InformatiQ committed:
Merge branch 'geobeau-master' into dir_mode_and_compressed

geobeau committed:
added filename select in URL

Added the ability to search for a filename and a full path in the URL argument select

geobeau committed:
Refactored

InformatiQ committed:
rename dir-mode to live-view

InformatiQ committed:
live view option in UI

geobeau committed:
Merge pull request #1 from geobeau/dir_mode_and_compressed

Dir mode and compressed

InformatiQ committed:
use sift if available instead of zcat|grep

geobeau committed:
Merge pull request #2 from criteo-forks/dir_mode_and_compressed

use sift if available instead of zcat|grep

InformatiQ committed:
unify output of tail and grep commands

geobeau committed:
Changed toolbar live-view position

geobeau committed:
Changed toolbar live-view position

geobeau committed:
Removed command select

geobeau committed:
Removed command select

InformatiQ committed:
make fn always an array to avoid discrepencies between different commands

geobeau committed:
Download URL from config

InformatiQ committed:
Merge branch 'geobeau-master' into dir_mode_and_compressed

geobeau committed:
Changed path/to/filename to filemane in download

InformatiQ committed:
add download-url config parameter

InformatiQ committed:
Merge branch 'geobeau-master' into dir_mode_and_compressed

InformatiQ committed:
append app name to download-url instead of ?app=

geobeau committed:
Enabling CORS for websockets

iksaif committed:
Merge pull request #9 from geobeau/patch-1

Enabling CORS for websockets

geobeau committed:
Added option to limit the number of greped lines

iksaif committed:
Merge pull request #10 from geobeau/patch-1

Added option to limit the number of greped lines

geobeau committed:
Added possibility to declare custom headers in configuration

Thib17 committed:
Merge pull request #11 from geobeau/patch-1

Added possibility to declare custom headers in configuration

geobeau committed:
Added EOF message to keep track of empty files

Thib17 committed:
Merge pull request #12 from geobeau/patch-1

Added EOF message to keep track of empty files
